### ISSUE TITLE
How to train from scratch

### ISSUE THREAD

**Issue by Electro1111** [lindawangg/COVID-Net#92]
Hello,

I noticed there is no model definition in this repository. I also noticed that this was pre-trained on image-net dataset. I was wondering if it would be possible to get the image-net pre-trained weights so that I could retrain this network on the dataset.

**Comment by haydengunraj** [comment]
Hi there,

With respect to the model definition, while we do not provide code for constructing the model, we do provide the model's metagraph file and trained weights. You can train the model from scratch by simply commenting out the lines in the training script where the weights are loaded.

To answer your question about ImageNet weights, unfortunately providing them will not be possible. Quoting the COVID-Net paper:

"The initial COVID-Net prototype was built and evaluated using the Keras deep learning library with a TensorFlow backend. The proposed COVID-Net architecture was built using generative synthesis, as described in Section 3.3."

What this means is that the prototype network was trained on ImageNet and COVIDx, but was then heavily modified during the generative synthesis process. As such, ImageNet weights for the final COVID-Net model do not exist - only the prototype network has ImageNet weights.

**Comment by Electro1111** [comment]
ok so to train from scratch would it just be a matter of commenting out this line in the code?

#saver.restore(sess, os.path.join(args.weightspath, args.ckptname))

**Comment by haydengunraj** [comment]
Yes, that's correct.

**Comment by Electro1111** [comment]
also from what I understand, this is the portion of the code that loads the metagraph, correct?:

tf.get_default_graph()
saver = tf.train.import_meta_graph(os.path.join(args.weightspath, args.metaname))
graph = tf.get_default_graph()

**Comment by haydengunraj** [comment]
That's right - tf.train.import_meta_graph() constructs the metagraph in the default graph.

**Comment by Electro1111** [comment]
Ok, thank you

are there any plans to release the scripts used to do the generative synthesis from the prototype model?

**Comment by haydengunraj** [comment]
Generative synthesis is part of the GenSynth deep learning development platform provided by [DarwinAI](https://www.darwinai.com/), and so we will not be able to release any code for this.

**Comment by Electro1111** [comment]
ahh ok. Has there been any experimentation with training the developed model from scratch and comparing the performance to training the model using the GenSynth?

**Comment by Electro1111** [comment]
I would also like to know if there is a recommended batch size, learning rate, number of epochs, etc when training the model from scratch because I do not see any info in the paper about these parameters.

**Comment by haydengunraj** [comment]
As far as I know this has not been explored - if you do so, we'd love to hear the results you achieve and settings you use!

**Comment by Electro1111** [comment]
One final question, was your model pretrained using a black and white version of Imagenet since these images are single channel?
